### PR TITLE
re-feat(provider): guilded provider added

### LINKED ISSUE #4382: [NV-2962] 🚀 Feature: Add Guilded Chat Provider
### 🔖 Feature description

Use the [Guilded docs](https://www.guilded.gg/docs/api/chat/ChatMessage) to create a new Chat provider while following [Novu's guide to a new provider](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider)

### 🎤 Why is this feature needed ?

provide the community with more providers

### ✌️ How do you aim to achieve this?

community contributions

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-2962](https://linear.app/novu/issue/NV-2962/🚀-feature-add-guilded-chat-provider)</sub>

### PR DESCRIPTION
**Pull request by Kanav-Arora**
### What change does this PR introduce?

Added Guilded chat provider as requested in #4382 

### Why was this change needed?

Closes #4382 

### Other information (Screenshots)

Screenshot 2023-10-11 at 10 49 10 PM

Screenshot 2023-10-11 at 10 49 24 PM

### CODE REVIEW

**Comment by Kanav-Arora** [comment]
Hi @Cliftonz 
Please review this

**Review by Cliftonz — CHANGES_REQUESTED** [review]
Dont forget your unit tests

File: providers/guilded/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Novu Guilded Provider
+
+A Guilded chat provider library for [@novu/node](https://github.com/novuhq/novu)
+
+## Usage
+
+```javascript
+    FILL IN THE INITIALIZATION USAGE
```
[the remark this answers is not in the document]
Comment (Kanav-Arora) [reply providers/guilded/README.md]: Done

**Comment by Kanav-Arora** [comment]
Hi I have made the requested changes
Just to confirm is provider.spec.ts the unit test file. Please help me with it

**Comment by Cliftonz** [comment]
> Hi I have made the requested changes Just to confirm is provider.spec.ts the unit test file. Please help me with it

I would recommend looking at other providers or posting in the issue that you need help from other commutiy members.

**Comment by Cliftonz** [comment]
@Kanav-Arora looks like there are a few conflicts that need resolved.

**Comment by Kanav-Arora** [comment]
Hi 
I have resolved merge conflict

**Comment by Cliftonz** [comment]
@scopsy Why is this PR showing over 150 commits?
I must be missing something

**Comment by Cliftonz** [comment]
@Kanav-Arora Please fix the conflicts as I think we are getting close to merging this

**Review by Cliftonz — APPROVED** [review]
Fix the conflicts and we are ready to go

**Comment by Kanav-Arora** [comment]
@Cliftonz Done 
Please merge

**Comment by Cliftonz** [comment]
@Kanav-Arora Did you purposely delete the repository?

**Pull request closed without merging**